### PR TITLE
can_summarize: Fix problems when signal type is given as qualified name

### LINKED ISSUE #148: Signal type specified as qualified name crashes the widget class
**Describe the bug**
I an on widget-base, canvas-core, orange, orange3-text master branches. I do not have orange3-network installed.

```
2021-04-09 08:49:15,566:WARNING:orangecanvas.registry.discovery: Error while importing 'orangecontrib.text.widgets.owcorpustonetwork'.
Traceback (most recent call last):
  File "/home/marko/dev/orange-widget-base/orangewidget/utils/signals.py", line 58, in can_summarize
    summarizer = summarize.dispatch(a_type)
  File "/usr/lib/python3.8/functools.py", line 828, in dispatch
    impl = dispatch_cache[cls]
  File "/usr/lib/python3.8/weakref.py", line 383, in __getitem__
    return self.data[ref(key)]
TypeError: cannot create weak reference to 'str' object

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/marko/dev/orange-canvas-core/orangecanvas/registry/discovery.py", line 260, in iter_widget_descriptions
    module = asmodule(name)
  File "/home/marko/dev/orange-canvas-core/orangecanvas/registry/discovery.py", line 503, in asmodule
    return __import__(module, fromlist=[""])
  File "/home/marko/dev/orange3-text/orangecontrib/text/widgets/owcorpustonetwork.py", line 45, in <module>
    class OWCorpusToNetwork(OWWidget, ConcurrentWidgetMixin):
  File "/home/marko/dev/orange3-text/orangecontrib/text/widgets/owcorpustonetwork.py", line 62, in OWCorpusToNetwork
    class Outputs:
  File "/home/marko/dev/orange3-text/orangecontrib/text/widgets/owcorpustonetwork.py", line 63, in Outputs
    network = Output("Network", "orangecontrib.network.Network")
  File "/home/marko/dev/orange-widget-base/orangewidget/utils/signals.py", line 228, in __init__
    self.auto_summary = auto_summary and can_summarize(type, name)
  File "/home/marko/dev/orange-widget-base/orangewidget/utils/signals.py", line 60, in can_summarize
    warnings.warn(f"{a_type.__name__} cannot be summarized. {instr}",
AttributeError: 'str' object has no attribute '__name__'
A clear and concise description of what the bug is.
```

### PR DESCRIPTION
**Pull request by janezd**
##### Issue

Fixes #148. @markotoplak, please confirm.

auto_summarize tried to issue a warning when the signal type was given as fully qualified name (a `str`), but crashed when composing the message.

##### Description of changes

Instead of catching a `TypeError` (raised by dispatcher when called by a string instead of a type), it now checks whether the type is given as a string and shows a better warning (without crashing).

`auto_summary` is disabled in this case, unless manually enabled in the signal definition. Function `can_summarize` could, in principle, check whether there is a summarizer declared for this name, but it is possible that the summarizer would actually appear later. This would be confusing, so it's better to require explicit `auto_summarize` argument in this (rare) case.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

File: orangewidget/utils/signals.py [hunk 1]
```diff
@@ -159,10 +161,10 @@ def set_train_data(self, data):
 
     def __init__(self, name, type, id=None, doc=None, replaces=None, *,
                  multiple=False, default=False, explicit=False,
-                 auto_summary=True, closing_sentinel=None):
+                 auto_summary=None, closing_sentinel=None):
         flags = self.get_flags(multiple, default, explicit, False)
         super().__init__(name, type, "", flags, id, doc, replaces or [])
-        self.auto_summary = auto_summary and can_summarize(type, name)
+        self.auto_summary = can_summarize(type, name, auto_summary)
```
Comment (markotoplak) [inline orangewidget/utils/signals.py:167]: The docstring for auto_summary should be updated.

File: orangewidget/utils/signals.py [hunk 2]
```diff
@@ -52,18 +52,20 @@ def base_summarize(_) -> PartialSummary:
...
     for a_type in type_:
-        try:
-            summarizer = summarize.dispatch(a_type)
-        except TypeError:
-            warnings.warn(f"{a_type.__name__} cannot be summarized. {instr}",
-                          UserWarning)
+        if isinstance(a_type, str):
+            warnings.warn(
+                f"Output is specified with qualified name ({a_type}). "
+                "To enable auto summary, set auto_summarize to True. "
```
Comment (markotoplak) [inline orangewidget/utils/signals.py:65]: auto_summarize or auto_summary?

**Review by markotoplak — COMMENTED** [review]
Thanks, no errors anymore. I added some minor comments.

File: orangewidget/utils/signals.py (hunk 1 above)
Comment (janezd) [reply orangewidget/utils/signals.py:179]: Updated. Plus added a docstring for `summarize`.

File: orangewidget/utils/signals.py (hunk 2 above)
Comment (janezd) [reply orangewidget/utils/signals.py]: Fixed -- here and a few lines above, in the existing code.

**Pull request merged**